libs/foundry-dev-tools/src/foundry_dev_tools/resources/dataset.py

```diff
@@ -558,19 +559,34 @@ def save_dataframe(
         """
         with self.transaction_context(transaction_type=transaction_type):
             from foundry_dev_tools._optional.pandas import pd
+            from foundry_dev_tools._optional.polars import pl
 
             # TODO needed?
             # to be backwards compatible to most readers, that expect files
             # to be under spark/
             folder = str(round(time.time() * 1000)) if transaction_type == "APPEND" else "spark"
-            if not pd.__fake__ and isinstance(df, pd.DataFrame):
+            if not (pd.__fake__ or pl.__fake__) and isinstance(df, pd.DataFrame | pl.DataFrame):
```

Comment:
Could you instead do an elif with the polars condition (`elif not pl.__fake__ and isinstance(df, pl.DataFrame)`) and the polars write_parquet command instead of one if at the top and another in the body?

Reply:
Basicaly:

if pandas:
  ... (leave unchanged)
elif polars:
   ... (write_parquet code
else: # spark
  .... (use parquet_compression from common definion above)